pkg/scheduler/framework/plugins/volumezone/volume_zone.go

```diff
@@ -232,7 +346,7 @@ func (pl *VolumeZone) Filter(ctx context.Context, cs *framework.CycleState, pod
 	if !hasAnyNodeConstraint {
 		// The node has no zone constraints, so we're OK to schedule.
 		// This is to handle a single-zone cluster scenario where the node may not have any topology labels.
-		return nil
+		return nil, true
 	}
```

Comment:
I know it's not your fault though, please `return nil, true` in L341-342. We don't need hasAnyNodeConstraint.

Reply:
@sanposhiho 
I can't understand why `hasAnyNodeConstraint` is no longer necessary. Could you please explain your thoughts in detail?

In my understanding, first check if a Node has a `topologyLabel`, and if the Node doesn't have any `topologyLabel` then `hasAnyNodeConstraint` is set to `false`. In that case, all pods can be scheduled. 
However, if `return nil, true` at L341, it makes scheduling possible as long as the node has at least one `topologyLabel`. It is not correct logic.

Reply:
Right, nvm.